src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1685 @@
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [|else if (b)|]
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfHeaderSelection3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+[|        else if (b)
+|]        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfFullSelectionWithoutElseClause1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (b)
+        {
+        }|]
+        else
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+        else
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfFullSelectionWithoutElseClause2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+[|        else if (b)
+        {
+        }
+|]        else
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+        else
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfFullSelectionWithElseClause1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (b)
+        {
+        }
+        else
+        {
+        }|]
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfFullSelectionWithElseClause2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+[|        else if (b)
+        {
+        }
+        else
+        {
+        }
+|]    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfParenthesisSelection()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b[|)|]
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task NotMergedOnElseIfElseKeywordSelection()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [|else|] if (b)
+        {
+        }
+    }
```

Comment:
out of curiosity, why?

Reply:
Because it's not one of the cases I explicitly added support for. This doesn't seems important to me.